### PR TITLE
Sync class for searching for videos and playlists in channel

### LINKED ISSUE #56: Is there any way to get channel videos by channel id?

### PR DESCRIPTION
**Pull request by mytja**
Title says it all
This resolves #56

### CODE REVIEW

**Review by alexmercerind — COMMENTED** [review]
> If they want to get ALL videos without specific search, they can use https://github.com/martinblech/xmltodict and this url: https://www.youtube.com/feeds/videos.xml?channel_id={ID}

The library is made to stimulate the web requests made by the browser. There are specific requests that are made when browsing through particular channel's videos. We don't need to depend upon XML response. 

For getting channel videos, existing `Playlist` class's API can be taken as reference, it has field `hasMoreVideos`, using `getNextVideos` one can get more videos. However, this will also require the optional parmeter for "Sort By" when browsing through channel's videos.

Rest seems fine.

File: youtubesearchpython/search.py [hunk 1]
```diff
@@ -247,6 +247,83 @@ def __init__(self, query: str, limit: int = 20, language: str = 'en', region: st
...
+                    },
+                    "type": "video"
+                },
+            ]
+        }
+    '''
+
+    def __init__(self, query: str, browseId: str, language: str = 'en', region: str = 'US', searchPreferences: str = "EgZzZWFyY2g%3D"):
+        super().__init__(query, language, region, searchPreferences, browseId)
+        #self._getComponents(*self.searchMode)
```
[the remark this answers is not in the document]
Comment (mytja) [reply youtubesearchpython/search.py:325]: Sorry. I forgot to remove it from when i was testing

**Comment by mytja** [comment]
Thanks for your review. I will fix it soon

**Review by alexmercerind — APPROVED** [review]
LGTM! 🚀

**Comment by mytja** [comment]
I also fixed sync StreamURLFetcher and async one.
See here: https://github.com/pytube/pytube/issues/1021#issuecomment-864239448

StreamURLFetcher is now failing at age restricted videos. May you have a look at it??

I will merge commits now

**Pull request merged**